submissions/src/main/kotlin/gov/cdc/prime/reportstream/submissions/controllers/SubmissionController.kt

```diff
@@ -0,0 +1,185 @@
+    private val queueServiceClient: QueueServiceClient,
+    private val telemetryClient: TelemetryClient,
+    private val tableClient: TableClient,
+    @Value("\${azure.storage.container-name}") private val containerName: String,
+    @Value("\${azure.storage.queue-name}") private val queueName: String,
+) {
+    private val logger = LoggerFactory.getLogger(SubmissionController::class.java)
+
+    /**
+     * Submits a report.
+     *
+     * This endpoint accepts reports in HL7 V2 and FHIR NDJSON formats. The report data is uploaded
+     * to Azure Blob Storage, queued for further processing, and its metadata is stored in Azure Table Storage.
+     * A custom event is also tracked in Application Insights for monitoring.
+     *
+     * @param headers the HTTP headers of the request
+     * @param contentType the content type of the report (must be "application/hl7-v2" or "application/fhir+ndjson")
+     * @param clientId the ID of the client submitting the report. Should represent org.senderName
+     * @param data the report data
+     * @return a ResponseEntity containing the reportID, status, and timestamp
+     */
+    @PostMapping("/api/v1/reports", consumes = ["application/hl7-v2", "application/fhir+ndjson"])
+    fun submitReport(
+        @RequestHeader headers: Map<String, String>,
+        @RequestHeader("Content-Type") contentType: String,
+        @RequestHeader(value = "client_id", required = false) clientId: String?,
+        @RequestBody data: String,
+    ): ResponseEntity<*> {
+        val reportId = UUID.randomUUID()
+        val reportReceivedTime = OffsetDateTime.now()
+        val status = "Received"
+        logger.info("Received report submission: reportId=$reportId, contentType=$contentType, clientId=$clientId")
+
+        try {
+
+            val headerValidationResult = validateHeaders(clientId)
+            if (headerValidationResult != null) {
+                logger.warn("Header validation failed: $headerValidationResult")
+                return headerValidationResult
+            }
+
+            // Convert data to ByteArray
+            val dataByteArray = data.toByteArray()
+            logger.debug("Converted report data to ByteArray")
+
+            // Upload to blob storage
+            val blobContainerClient = blobServiceClient.getBlobContainerClient(containerName)
+            val blobClient = blobContainerClient.getBlobClient(formBlobName(reportId, headers))
+            blobClient.upload(dataByteArray.inputStream(), dataByteArray.size.toLong())
+            logger.info("Uploaded report to blob storage: blobUrl=${blobClient.blobUrl}")
+
+            // Create the message for the queue
+            val message = mapOf(
+                "reportId" to reportId.toString(),
+                "blobUrl" to blobClient.blobUrl,
+                "headers" to filterHeaders(headers)
+            )
+            val objectMapper = jacksonObjectMapper().registerModule(JavaTimeModule())
```

Comment:
Maybe as a follow up, but we should just move the jackson utilities to the shared project so we don't need to make one again here